node/src/p2p.rs

```diff
@@ -426,13 +440,13 @@ where
                     self.report();
                 }
                 _ = kademlia_interval.tick() => {
-                    // Bootstrap procedure is a bit misleading as a name. It is actually
-                    // scanning the network thought the already known peers and find new
-                    // ones. It also recovers connectivity of previously known peers and
-                    // refreshes the routing table.
-                    //
-                    // libp2p team suggests to start bootstrap procedure every 5 minute.
-                    let _ = self.swarm.behaviour_mut().kademlia.bootstrap();
+                    if self.peer_tracker.info().num_connected_peers < MIN_CONNECTED_PEERS
+                        || kademlia_last_bootstrap.elapsed() > KADEMLIA_BOOTSTRAP_PERIOD
+                    {
+                        info!("Running kademlia bootstrap procedure.");
```

Comment:
I think I would log this as debug, but up to you.

Reply:
done